zipline/data/minute_bars.py

```diff
@@ -415,11 +413,49 @@ def write(self, sid, df):
                 volume : float64|int64
             index : DatetimeIndex of market minutes.
         """
+        cols = {
+            'open': df.open.values,
+            'high': df.high.values,
+            'low': df.low.values,
+            'close': df.close.values,
+            'volume': df.volume.values,
+        }
+        dts = df.index.values
```

Comment:
(Summarizing conversation in another channel.)

'internal' meaning an 'internal repo'.
Algos would not be calling this function, it would part of the one-time setup, and it would most likely be the case that this would be passed `DataFrame`s that cover large spans of data.